src/test/scala/nl/knaw/dans/easy/stage/EasyStageDatasetSpec.scala

```diff
@@ -16,156 +16,144 @@
+class EasyStageDatasetSpec extends FlatSpec with Matchers with OneInstancePerTest {
+  private val testDir = Paths.get("target/test", getClass.getSimpleName)
+  FileUtils.deleteQuietly(testDir.toFile)
+  Files.createDirectories(testDir)
 
-  private val tmpProps = new File("cfg/application.properties")
+  private val testBagMedium = Paths.get("src/test/resources/dataset-bags/medium")
 
   "createFileAndFolderSdos" should "stumble if the data folder does not exist" in {
+    val bagDir = testDir.resolve("bag")
+    val dataDir = bagDir.resolve("data") // Note that this directory is never created, on purpose.
+    val sdoSetDir = testDir.resolve("someSDO")
+    implicit val s = createSettings(bagDir.toFile, sdoSetDir.toFile)
+    Files.createDirectories(bagDir)
 
-    val bagitDir = new File("target/test/bag")
-    val dataDir = new File(bagitDir, "data")
-    val sdoSetDir = new File("target/test/someSDO")
-    implicit val s = createSettings(bagitDir, sdoSetDir)
-    bagitDir.mkdirs()
-
-    createFileAndFolderSdos(dataDir, DATASET_SDO, ANONYMOUS_ACCESS) shouldBe a[Failure[_]]
-
-    deleteDirectory(bagitDir)
+    createFileAndFolderSdos(dataDir.toFile, DATASET_SDO, ANONYMOUS_ACCESS) shouldBe a[Failure[_]]
   }
 
   it should "be happy with an empty data folder" in {
-
-    val bagitDir = new File("target/test/bag")
-    val dataDir = new File(bagitDir, "data")
-    val sdoSetDir = new File("target/test/someSDO")
-    implicit val s = createSettings(bagitDir, sdoSetDir)
-    dataDir.mkdirs()
-
-    createFileAndFolderSdos(dataDir, DATASET_SDO, ANONYMOUS_ACCESS) shouldBe a[Success[_]]
-    sdoSetDir.exists() shouldBe false
-
-    deleteDirectory(bagitDir)
+    val bagDir = testDir.resolve("bag")
+    val dataDir = bagDir.resolve("data")
+    val sdoSetDir = testDir.resolve("someSDO")
+    implicit val s = createSettings(bagDir.toFile, sdoSetDir.toFile)
+    Files.createDirectories(dataDir)
+
+    createFileAndFolderSdos(dataDir.toFile, DATASET_SDO, ANONYMOUS_ACCESS) shouldBe a[Success[_]]
+    Files.exists(sdoSetDir) shouldBe false
   }
 
   it should "stumble over a manifest-sha1.txt with too many fields on a line" in {
-
-    val bagitDir = new File("target/test/bag")
-    val dataDir = new File(bagitDir, "data")
-    val sdoSetDir = new File("target/test/someSDO")
-    implicit val s = createSettings(bagitDir, sdoSetDir)
-    dataDir.mkdirs()
-    FileUtils.write(new File(bagitDir,"manifest-sha1.txt"),"a b c")
-
-    createFileAndFolderSdos(dataDir, DATASET_SDO, ANONYMOUS_ACCESS).failed.get should
+    val bagDir = testDir.resolve("bag")
+    val dataDir = bagDir.resolve("data")
+    val sdoSetDir = testDir.resolve("someSDO")
+    implicit val s = createSettings(bagDir.toFile, sdoSetDir.toFile)
+    Files.createDirectories(dataDir)
+    FileUtils.write(bagDir.resolve("manifest-sha1.txt").toFile,"a b c")
+
+    createFileAndFolderSdos(dataDir.toFile, DATASET_SDO, ANONYMOUS_ACCESS).failed.get should
```

Comment:
Mmm, didn't know you could do `have message` on a `Throwable`. Very nice! But then why not use an even more idiomatic `ScalaTest`-like style? I don't really like the `.failed.get` here 😛 

```scala
(the [IllegalArgumentException] thrownBy createFileAndFolderSdos(dataDir.toFile, DATASET_SDO, ANONYMOUS_ACCESS).get) should have message "Invalid line in manifest-sha1.txt: a b c"
```
or

```scala
inside(createFileAndFolderSdos(dataDir.toFile, DATASET_SDO, ANONYMOUS_ACCESS)) {
  case Failure(e: IllegalArgumentException) => e should have message "Invalid line in manifest-sha1.txt: a b c"
}
```